From 9c394f2e0338cb37c7192c3436c2b611b1efb3c3 Mon Sep 17 00:00:00 2001
From: Ren Chenglei <chenglei.ren@intel.com>
Date: Fri, 9 Jun 2023 21:48:08 +0530
Subject: [PATCH] INTERNAL: Add camera usage for RGBA format

On virgl backend, we use vhal camera. During camera
preview, we use RGBA format, so add camera usage for
it.

Tracked-On: OAM-96725
Signed-off-by: Ren Chenglei <chenglei.ren@intel.com>
---
 virtgpu_virgl.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/virtgpu_virgl.c b/virtgpu_virgl.c
index 9474c40..b84e40a 100644
--- a/virtgpu_virgl.c
+++ b/virtgpu_virgl.c
@@ -668,6 +668,10 @@ static int virgl_init(struct driver *drv)
 					   BO_USE_HW_VIDEO_DECODER | BO_USE_HW_VIDEO_ENCODER);
 	}
 
+#ifdef USE_GRALLOC1
+	drv_modify_combination(drv, DRM_FORMAT_ABGR8888, &LINEAR_METADATA,
+			       BO_USE_CAMERA_WRITE | BO_USE_CAMERA_READ);
+#endif
 	return drv_modify_linear_combinations(drv);
 }
 
-- 
2.40.1

